### ISSUE TITLE
Case sensitivity mismatch leads to "Required module not found" errors

### ISSUE THREAD

**Issue by jamesisaac** [facebook/flow#3125]
This is related to #2117 but was a lot more of a pain to debug, as the errors it results in provide no clue that it's a case sensitivity issue.

One day out of nowhere I was suddenly getting hundreds of `Required module not found` errors when running a flow check, both on third party modules and import references to my own code.  Tried reinstalling all npm modules etc with no luck.

``` js
navigation/RootNavigation.js:8
  8: } from 'react-native'
            ^^^^^^^^^^^^^^ react-native. Required module not found

navigation/RootNavigation.js:22
 22: import Colors from '../constants/Colors'
                        ^^^^^^^^^^^^^^^^^^^^^ ../constants/Colors. Required module not found
```

Turns out I'd just happened to `cd` to the path of the project lazily using lowercase paths (works fine in the git bash in Windows 10), when there are a couple of uppercase letters in the real path.  Unlike the examples in #2117 , I wasn't even getting a client/server mismatch error, as my first time running the server was from the lazily-cased path.

Flow simply refusing to type check and erroring out with a message that the current path is incorrect (similar to the referenced issue) would obviously be preferable to a load of errors about module resolution, which are a bit of a red herring as they only mention relative paths.

Appears in 0.33.0 and 0.37.4 (latest).

**Comment by mroch** [comment]
You mention Windows 10, is that where this is happening? Linux has case sensitive file systems, OS X is insensitive, and I'm not sure what Windows uses.

**Comment by jamesisaac** [comment]
Yeah, this is on Windows 10, with the Git Bash (comes bundled with Git for Windows).  Just tried with command prompt and that does force the correct casing in the current working dir, so it's not a problem there.  But I think it's quite a common setup on Windows to be using these tools that mimic the Linux command line (Cygwin etc).

**Comment by mroch** [comment]
yeah, and if it also repros on OS X that'd also be a very common setup

**Comment by yassinecc** [comment]
@mroch Same issue on OS X
